subsys/mpsl/Kconfig

```diff
@@ -25,6 +25,15 @@ config MPSL_TIMESLOT_SESSION_COUNT
 	help
 	  Maximum number of timeslot sessions.
 
+config MPSL_SCHEDULER_INIT
+	bool "Initialize MPSL scheduler"
```

Comment:
I would change this option to not mention the "init" part, that seems like a technicality.
config MPSL_SCHEDULER 
    bool "MPSL scheduler"

Does this option have to be available to the user?

The dependencies are not very well handled now I think. The user could end up with the invalid configuration of disabling SCHEDULER and enabling BT.

There is no way for the application to use the scheduler without timeslot AFAIK.
I would suggest to do this:
```
config MPSL_SCHEDULER_INIT
    bool
    default y if TIMESLOT_COUNT > 0
```
And adding select to the other options that need it, like the SoftDevice controller.

For Zigbee which has it as optional It would be better to imply this option.
And using that as the switch that enables zigbee to schedule using MPSL, if that makes sense.

Reply:
It seems like a good idea to me. So I renamed this option to MPSL_SCHEDULER and made it unavailable to the user. I added the select statement to SoftDevice Controller and an imply  statement to Zibee.